### ISSUE TITLE
[3.1.2] Build warnings in tools used when compiling using rules_swift

### ISSUE THREAD

**Issue by csmulhern** [bazelbuild/rules_swift#1557]
```bash
INFO: From Compiling Swift module @@rules_swift+//tools/test_observer:test_observer:
external/rules_swift+/tools/test_observer/SwiftTestingRunner.swift:83:26: warning: result of call to 'withLock' is unused
 81 |       switch testOrSuite {
 82 |       case .suite(let suiteID):
 83 |         discoveredSuites.withLock { $0.insert(suiteID) }
    |                          `- warning: result of call to 'withLock' is unused
 84 |       case .test(let test):
 85 |         collector.addTest(test)

external/rules_swift+/tools/test_observer/SwiftTestingRunner.swift:179:34: warning: trailing closure in this context is confusable with the body of the statement; pass as a parenthesized argument to silence this warning
177 |       case .string(let testID) = payload["testID"],
178 |       // Ignore suites. The xUnit recorder reconstructs the hierarchy.
179 |       !discoveredSuites.withLock { $0.contains(testID) },
    |                                  `- warning: trailing closure in this context is confusable with the body of the statement; pass as a parenthesized argument to silence this warning
180 |       case .object(let instantJSON) = payload["instant"],
181 |       case .number(let absolute) = instantJSON["absolute"]

INFO: From Compiling Swift module @@rules_swift++non_module_deps+com_github_apple_swift_docc_symbolkit//:SymbolKit [for tool]:
external/rules_swift++non_module_deps+com_github_apple_swift_docc_symbolkit/Sources/SymbolKit/UnifiedSymbolGraph/UnifiedSymbolGraph+Encodable.swift:44:30: warning: 'relationships' is deprecated: Use unifiedRelationships and orphanRelationships instead
42 |
43 |         try container.encode(Array(symbols.values), forKey: .symbols)
44 |         try container.encode(relationships, forKey: .relationships)
   |                              `- warning: 'relationships' is deprecated: Use unifiedRelationships and orphanRelationships instead
45 |     }
46 | }
```

The last one appears to be out of repo, but the first two are from `//tools/test_observer`.
